metadata-integration/java/datahub-client/src/main/java/datahub/client/patch/dataset/DatasetPropertiesPatchBuilder.java

```diff
@@ -1,76 +1,119 @@
+          instance.textNode(externalUrl)));
+    }
+    return this;
   }
 
-  @Override
-  protected JsonNode getValue() {
-    ObjectNode value = instance.objectNode();
-
-    if (customProperties != null) {
-      value.set(CUSTOM_PROPERTIES_KEY, OBJECT_MAPPER.valueToTree(customProperties));
-    }
-    if (description != null) {
-      value.put(DESCRIPTION_KEY, description);
+  public DatasetPropertiesPatchBuilder setName(@Nullable String name) {
+    if (name == null) {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(),
+          BASE_PATH + NAME_KEY, null));
+    } else {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), BASE_PATH + NAME_KEY, instance.textNode(name)));
     }
-    if (uri != null) {
-      value.put(URI_KEY, uri);
-    }
-    if (qualifiedName != null) {
-      value.put(QUALIFIED_NAME_KEY, qualifiedName);
-    }
-    if (name != null) {
-      value.put(NAME_KEY, name);
+    return this;
+  }
+
+  public DatasetPropertiesPatchBuilder setQualifiedName(@Nullable String qualifiedName) {
+    if (qualifiedName == null) {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(),
+          BASE_PATH + QUALIFIED_NAME_KEY, null));
+    } else {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), BASE_PATH + QUALIFIED_NAME_KEY,
+          instance.textNode(qualifiedName)));
     }
-    if (externalUrl != null) {
-      value.put(EXTERNAL_URL_KEY, externalUrl);
+    return this;
+  }
+
+  public DatasetPropertiesPatchBuilder setDescription(@Nullable String description) {
+    if (description == null) {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(),
+          BASE_PATH + DESCRIPTION_KEY, null));
+    } else {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), BASE_PATH + DESCRIPTION_KEY,
+          instance.textNode(description)));
     }
-    if (tags != null) {
-      value.put(TAGS_KEY, OBJECT_MAPPER.valueToTree(tags));
+    return this;
+  }
+
+  public DatasetPropertiesPatchBuilder setUri(@Nullable String uri) {
+    if (uri == null) {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(),
+          BASE_PATH + URI_KEY, null));
+    } else {
+      this.pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), BASE_PATH + URI_KEY, instance.textNode(uri)));
     }
+    return this;
+  }
+
+  @Override
+  public DatasetPropertiesPatchBuilder addCustomProperty(@Nonnull String key, @Nonnull String value) {
+    this.customPropertiesPatchBuilder.addProperty(key, value);
+    return this;
+  }
 
-    return value;
+  @Override
+  public DatasetPropertiesPatchBuilder removeCustomProperty(@Nonnull String key) {
+    this.customPropertiesPatchBuilder.removeProperty(key);
+    return this;
+  }
+
+  @Override
+  public DatasetPropertiesPatchBuilder replaceCustomProperties(Map<String, String> properties) {
```

Comment:
change to `setCustomProperties`

Reply:
write javadoc for what this does and when to use it versus addCustomProperty / removeCustomProperty